inc/hclib-async.h

```diff
@@ -198,6 +217,26 @@ inline void async_nb_await_at(T&& lambda, hclib_future_t *fut,
     spawn_await_at(task, fut ? &fut : NULL, fut ? 1 : 0, locale);
 }
 
+template <typename T>
+inline void async_nb_await_at(T&& lambda, std::vector<hclib_future_t *> &futures,
```

Comment:
Is there a reason futures has type std::vector<hclib_future_t *>& here and std::vector<hclib_future_t *>&& in other places?

Reply:
vector<hclib_future_t *> fut_vec{f1, f2, f3}; async_await(...,fut_vec); will invoke lvalue reference whereas,

async_await(..., vector<hclib_future_t *>{f1, f2, f3}); would invoke rvalue reference. If function with &&futures is not defined,  it will give compile time since there is no function matching this signature.